pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -71,66 +80,106 @@ func (q *QueryReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctr
 		return res, err
 	}
 
-	return ctrl.Result{}, q.runListRestorePoints(ctx, req)
+	// setup dispatcher for vclusterops API
+	dispatcher, err := q.makeDispatcher(q.Log, q.Vdb, nil /*password*/)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	finder := iter.MakeSubclusterFinder(q.VRec.Client, q.Vdb)
+	pods, err := finder.FindPods(ctx, iter.FindExisting)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// find a pod to execute the vclusterops API
+	podIP, err := q.findRunningPodWithNMAContainer(pods)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// extract out the communal and config information to pass down to the vclusterops API.
+	opts := []restorepointsquery.Option{}
+	opts = append(opts,
+		restorepointsquery.WithInitiator(q.Vrpq.ExtractNamespacedName(), podIP),
+		restorepointsquery.WithCommunalPath(q.Vdb.GetCommunalPath()),
+		restorepointsquery.WithConfigurationParams(q.ConfigurationParams.GetMap()),
+	)
+	return ctrl.Result{}, q.runShowRestorePoints(ctx, dispatcher, opts)
 }
 
-// fetch the VerticaDB and collect access information to the communal storage for the VerticaRestorePointsQuery CR,
-func (q *QueryReconciler) collectInfoFromVdb(ctx context.Context) (ctrl.Result, error) {
+// fetch the VerticaDB and collect access information to the communal storage for the VerticaRestorePointsQuery CR
+func (q *QueryReconciler) collectInfoFromVdb(ctx context.Context) (res ctrl.Result, err error) {
 	vdb := &v1.VerticaDB{}
-	res := ctrl.Result{}
-
-	opts := []restorepointsquery.Option{
-		restorepointsquery.WithInitiator(q.InitiatorPod, q.InitiatorPodIP),
+	if res, err = fetchVDB(ctx, q.VRec, q.Vrpq, vdb); verrors.IsReconcileAborted(res, err) {
```

Comment:
Why do we fetch the vdb? We should already have one present.

Reply:
This is the one present from https://jira.verticacorp.com/jira/browse/VER-90383. I just remove RetryOnConflict based on your previous comments

Reply:
As discussed, fetchVDB is still necessary because we require gathering information from VerticaDB; specifically, we need to retrieve fields in Vdb